### PR TITLE
Phase2 bughunt v4

### PR DESCRIPTION
**Pull request by Ronak-Searce**
- Add view assessment component.
- Illegal Name message for table added.

### CODE REVIEW

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.ts [hunk 1]
```diff
@@ -7,11 +10,74 @@ import { SidenavService } from 'src/app/services/sidenav/sidenav.service'
...
+        this.setRatePcDataSource()
+      }
+    })
+  }
 
   closeSidenav() {
     this.sidenav.closeSidenav()
   }
+  setDbDataSource() {
+    this.dbDataSource = []
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.ts:54]: Can we show database version as well as per figma ? : https://www.figma.com/file/mhSfts8ey0XxeHXyXdP1aC/HarbourBridge-UI?node-id=1179%3A42177

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.scss [hunk 2]
```diff
@@ -0,0 +1,82 @@
+.header {
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.scss:1]: Please place these css in styles.css as much as possible. We should avoid local css.

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 3]
```diff
@@ -1,2 +1,130 @@
...
+        <th mat-header-cell *matHeaderCellDef>
+          <mat-icon class="icon success icon-size icons"> check_circle </mat-icon>100% conversion
+        </th>
+        <td mat-cell  class="cells" *matCellDef="let element">{{ element.good }}</td>
+      </ng-container>
+
+      <tr mat-header-row *matHeaderRowDef="rateCountDisplayedColumns"></tr>
+      <tr mat-row *matRowDef="let row; columns: rateCountDisplayedColumns"></tr>
+    </table>
+    <h3>Conversion status by table percentage</h3>
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:68]: We don't need this conversion status by percentage , it is not providing any extra information.

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 4]
```diff
@@ -1,2 +1,130 @@
...
+          class="success-background"
+          [ngStyle]="{ 'width.%': this.conversionRatePercentage.good }"
+        ></div>
+      </div>
+      <hr />
+      <div class="percentage-marker">
+        <span>1%</span>
+        <span>50%</span>
+        <span>100%</span>
+      </div>
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:115]: these markers are confusing let's remove these

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 5]
```diff
@@ -1,2 +1,130 @@
...
+      <div class="percentage-indent">
+        <span
+          ><mat-icon class="icon danger"> circle </mat-icon>
+          <span>Not a great conversion</span></span
+        >
+        <span
+          ><mat-icon class="icon warning"> circle </mat-icon>
+          <span>Converted with warnings</span></span
+        >
+        <span><mat-icon class="icon success"> circle </mat-icon> <span>Converteds</span></span>
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:126]: Rename it to 100% conversion or converted automatically

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html (hunk 3 above)
Comment (Ronak-Searce) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html]: Updated

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html (hunk 4 above)
Comment (Ronak-Searce) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html]: Updated

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.scss (hunk 2 above)
Comment (mayank2913) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.scss]: Updated

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 6]
```diff
@@ -1,2 +1,101 @@
-<h2>Assessment content coming soon!</h2>
-<button mat-button (click)="closeSidenav()">Close</button>
+<div>
+  <div class="header">
+    <span class="mat-h3 header-title">Assessment report</span>
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:3]: https://drive.google.com/file/d/1PAMjagdSRO7ZHSy9ZzrQ0Zb2_MynDdO6/view?usp=sharing&resourcekey=0-idXhZV2UqPtTh_p2fm8u_Q The header is not of same size as in other pages

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 7]
```diff
@@ -1,2 +1,101 @@
...
+    <button class="close-button" mat-icon-button color="primary" (click)="closeSidenav()">
+      <mat-icon class="close-icon">close</mat-icon>
+    </button>
+  </div>
+  <div class="content">
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:8]: There is no left spacing between the dialog box and contents. The margin/padding needs to be fixed as in other components

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.ts [hunk 8]
```diff
@@ -7,11 +10,74 @@ import { SidenavService } from 'src/app/services/sidenav/sidenav.service'
...
+  setRateCountDataSource(tableCount: number) {
+    this.rateCountDataSource = []
+    this.rateCountDataSource.push({
+      total: tableCount,
+      bad: this.conversionRateCount.bad,
+      ok: this.conversionRateCount.ok,
+      good: this.conversionRateCount.good,
+    })
+  }
+  setRatePcDataSource() {
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.ts:75]: Do we need this now ?

Comment (mayank2913) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.ts]: Removed

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html (hunk 7 above)
Comment (mayank2913) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:8]: Updated the CSS

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html (hunk 6 above)
Comment (mayank2913) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html]: Updated the CSS

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 9]
```diff
@@ -1,2 +1,101 @@
...
+<div class="sidenav-view-assessment-container">
+  <div class="sidenav-view-assessment-header">
+    <span class="mat-h2 header-title">Assessment report</span>
+    <button class="close-button" mat-icon-button color="primary" (click)="closeSidenav()">
+      <mat-icon class="close-icon">close</mat-icon>
+    </button>
+  </div>
+  <div class="content">
+    <p>
+      Estimation for {{ srcDbType.toUpperCase() }} to for Spanner conversion:
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:10]: Estimation for {{ srcDbType.toUpperCase() }} to Spanner conversion:

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html [hunk 10]
```diff
@@ -1,2 +1,101 @@
...
+      {{ conversionRatePercentage.good }}% of tables can be converted automatically,
+      {{ conversionRatePercentage.ok }}% requires minimal conversion changes and
+      {{ conversionRatePercentage.bad }}% requires high complexity conversion changes.
+    </p>
+
+    <p></p>
+    <h3 class="sidenav-title">Source and destination database definitions</h3>
+    <table mat-table [dataSource]="dbDataSource" >
```
Comment (shreyakhajanchi) [inline ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:18]: The spacing between the header and table component should be same as the spacing between 'Conversion status by table' and the content below it : https://drive.google.com/file/d/16ACZFFcPhc4e6n3HWXnNiDIg8GURmVNQ/view?usp=share_link&resourcekey=0-gXW5_HRqxun4_M02YvquKA

**Review by shreyakhajanchi — COMMENTED** [review]
Overall LGTM, left minor comments

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html (hunk 9 above)
Comment (Ronak-Searce) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html:10]: Updated

File: ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html (hunk 10 above)
Comment (Ronak-Searce) [reply ui/src/app/components/sidenav-view-assessment/sidenav-view-assessment.component.html]: Updated

**Review by shreyakhajanchi — APPROVED** [review]
LGTM

**Pull request merged**